internal/api_server/agentserver/grpc.go

```diff
@@ -43,11 +48,51 @@ func NewAgentGrpcServer(
 		pendingStreams: &sync.Map{},
 	}
 }
+func validateClientTlsCert(ctx context.Context) (context.Context, error) {
```

Comment:
Can we as a follow up move validateClientTlsCert, and authFn to the API server middleware (this one in tls, the other one probably another .go file) ?

Reply:
Sure, i will do that in a followup.

Reply:
Followup https://github.com/flightctl/flightctl/pull/435